### PR TITLE
Remove trailing slash from dir paths before checking uniqueness. Fix #519

### LINKED ISSUE #519: Standardpath uses same directory more than once
### Steps to reproduce
Use ubuntu 22.04

> $ echo $XDG_DATA_DIRS
> /usr/share/ubuntu:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop

Trailing '/' is not a good practice, but this is what is given from this distribution.

In addons, use `standardPath::scandirectories(StandardPath::Type::Data, callback)`

### Expected
The callback is called by unique paths.

### Actual
The callback is called with `/usr/local/share` and `/usr/local/share/`, same in the sense of path.

### PR DESCRIPTION
**Pull request by naokiri**
Remove trailing slash from dir paths before checking uniqueness. This fixes fcitx#519

### CODE REVIEW

File: src/lib/fcitx-utils/fs.cpp [hunk 1]
```diff
@@ -143,6 +143,14 @@ std::string cleanPath(const std::string &path) {
     return buf;
 }
 
+std::string cleanDirPath(const std::string &dirPath) {
+    auto cleanedPath = cleanPath(dirPath);
+    while (!cleanedPath.empty() && cleanedPath.back() == '/') {
```
Comment (wengxt) [inline src/lib/fcitx-utils/fs.cpp:148]: This would be wrong if dirPath contains only slash.

**Review by wengxt — CHANGES_REQUESTED** [review]
Please also add test case to this function.

**Comment by wengxt** [comment]
Actually I believe there's no need to use cleanDirPath. cleanPath can already remove trailing slash.

**Comment by naokiri** [comment]
You're right the cleanPath already did the job. Thanks for fixing!

**Pull request merged**